packages/ui-source-code-editor/src/SourceCodeEditor/SearchPanel.tsx

```diff
@@ -0,0 +1,136 @@
+ * SOFTWARE.
+ */
+
+import React, { useState } from 'react'
+import { createRoot } from 'react-dom/client'
+import {
+  setSearchQuery,
+  search,
+  findNext,
+  findPrevious,
+  SearchQuery,
+  closeSearchPanel
+} from '@codemirror/search'
+import { EditorView } from '@codemirror/view'
+import { TextInput } from '@instructure/ui-text-input'
+import { IconButton } from '@instructure/ui-buttons'
+import {
+  IconArrowOpenDownLine,
+  IconArrowOpenUpLine,
+  IconSearchLine
+} from '@instructure/ui-icons'
+
+function SearchPanel({ view }: { view: EditorView }) {
+  const [searchQueryStr, setSearchQueryStr] = useState('')
+
+  const handleChange = (
+    _e: React.ChangeEvent<HTMLInputElement>,
+    value: string
+  ) => {
+    setSearchQueryStr(value)
+    handleHighlightSearch(value)
+  }
+
+  const handleHighlightSearch = (searchStr: string) => {
+    view.dispatch({
+      effects: setSearchQuery.of(
+        new SearchQuery({
+          search: searchStr
+        })
+      )
+    })
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key !== 'Enter') return
+    if (!e.shiftKey) handleFindNext()
+    else handleFindPrev()
+  }
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key !== 'Escape') return
+    closeSearchPanel(view)
+  }
+
+  const handleFindNext = () => {
+    handleHighlightSearch(searchQueryStr)
+    findNext(view)
+  }
+
+  const handleFindPrev = () => {
+    handleHighlightSearch(searchQueryStr)
+    findPrevious(view)
+  }
+
+  return (
+    <div className="search-panel">
+      <TextInput
+        inputRef={(r) => r?.focus()}
+        size="small"
+        display="inline-block"
+        width="20rem"
+        placeholder="Find in code..."
+        onChange={handleChange}
+        onKeyDown={handleKeyDown}
+        onKeyUp={handleKeyUp}
+        renderBeforeInput={<IconSearchLine size="x-small" />}
+        renderAfterInput={
+          <span>
+            <IconButton
+              size="small"
+              withBorder={false}
+              withBackground={false}
+              onClick={handleFindNext}
+              screenReaderLabel="Next result"
+            >
+              <IconArrowOpenDownLine />
+            </IconButton>
+            <IconButton
+              size="small"
+              withBorder={false}
+              withBackground={false}
+              onClick={handleFindPrev}
+              screenReaderLabel="Previous result"
+            >
+              <IconArrowOpenUpLine />
+            </IconButton>
+          </span>
+        }
+      />
+    </div>
+  )
+}
+
+export default function customSearch() {
+  return search({
+    createPanel: (view) => {
+      const dom = document.createElement('div')
+      dom.style.padding = '8px'
+      const root = createRoot(dom)
+      root.render(<SearchPanel view={view} />)
+      return { dom }
+    },
+    top: true // this is not working if the `createPanel` prop is present. might be a bug on codemirror's side
```

Comment:
hmm in this case should not we remove it? Codemirror might fix the bug and suddenly search panels will appear at the top..

Reply:
good point, I will remove it